datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -610,6 +610,35 @@ pub fn to_substrait_agg_measure(
                 }
             })
         }
+        Expr::AggregateUDF(expr::AggregateUDF{ fun, args, filter, order_by }) =>{
+            let sorts = if let Some(order_by) = order_by {
+                order_by.iter().map(|expr| to_substrait_sort_field(expr, schema, extension_info)).collect::<Result<Vec<_>>>()?
+            } else {
+                vec![]
+            };
+            let mut arguments: Vec<FunctionArgument> = vec![];
+            for arg in args {
+                arguments.push(FunctionArgument { arg_type: Some(ArgType::Value(to_substrait_rex(arg, schema, 0, extension_info)?)) });
+            }
+            let function_name = fun.name.to_lowercase();
```

Comment:
Why lowercase?

Reply:
I don't have a specific reason, and I guess this is a convention from translating SQL ident, where strings are preprocessed with `normalize_ident()` before mapping to concrete functions. But at this stage the input is all logical plans. `to_lowercase()` seems unnecessary. How do you think about it?

Reply:
Update: I found that ' to_lowercase` is unnecessary since `_register_function()` will do that. I've removed all `to_lowercase` in `producer.rs` except the one in `register_function()`

Reply:
Thank you @waynexia